webrtc.html

```diff
@@ -2582,6 +2636,110 @@
       </dl>
     </section>
 
+    <section>
+      <h2 id="sec.cert-mgmt">Certificate Management</h2>
+      <p>
+        The certificates that <code>RTCPeerConnection</code> instances use to
+        authenticate with peers use the <a><code>RTCCertificate</code></a>
+        interface.  These objects can be explicitly generated by applications
+        using <code>RTCPeerConnection.generateCertificate</code> and provided in
+        the <a><code>RTCConfiguration</code></a> when constructing a
+        new <code>RTCPeerConnection</code> instance.
+      </p>
+      <p>
+        The explicit certificate management functions provided here are
+        optional.  If an application does not provide
+        the <code>certificates</code> configuration option when constructing
+        an <code>RTCPeerConnection</code> a new set of certificates MUST be
+        generated by the <a>user agent</a>.  That set MUST include an ECDSA
+        certificate with a private key on the P-256 curve and a signature with a
+        SHA-256 hash.
+      </p>
+
+      <dl class="idl" title="partial interface RTCPeerConnection">
+        <dt>static Promise&lt;RTCCertificate&gt; generateCertificate
+          (AlgorithmIdentifier keygenAlgorithm)</dt>
+        <dd>
+          <p>This function takes an argument that describes how the private key associated with the certificate is generated.  The <var>keygenAlgorithm</var> argument uses the WebCrypto
+          [[!WebCryptoAPI]] <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#dfn-AlgorithmIdentifier">AlgorithmIdentifier</a>
+          type.  The <var>keygenAlgorithm</var> value MUST be a valid argument
+            to <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#SubtleCrypto-method-generateKey"><code>Crypto.subtle.generateKey</code></a>.</p>
```

Comment:
OK, let try to be clear: you are concerned about using duck-typing into WebCrypto concepts here.  I'm sorry that you feel that way, but I'm not inclined to back down on this. 

This is only duck-typing, not reuse.  I'm fully prepared to break any dependencies on WebCrypto by using copy and paste.  I can produce a duck made with all-new meat if necessary.  That's just more work for me.

Regarding specification style, I accept that you are explaining a method for getting to the same outcome. I'm happy to talk about how the approach you advocate produces bad work that fails to properly respect the Priority of Constituencies, but this isn't the time and place.